src/Middleware/RateLimiting/src/DefaultAcquiredLease.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.RateLimiting;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+internal sealed class DefaultAcquiredLease : RateLimitLease
+{
+    public override bool IsAcquired => true;
```

Comment:
I'm a little confused here. Shouldn't this be false? If it's true, does it mean that if the request is cancelled, then all the rate limiting policy for that request would be skipped and request would execute completely unless that somewhere in the pipeline `HttpContext.RequestAborted.IsCancellationRequested` is being checked?

Reply:
If this was false it would cause `OnRejected` to fire - but the request didn't get rejected by the RateLimiter, it got canceled. So in this case the middleware shouldn't take any action.

Reply:
We should check if we got this particular lease instance and no-op without calling through to the `next()` middleware and without calling `OnRejected`.

Reply:
Hmmm 🤔. Maybe check if the lease is this certain type and just return from the middleware.

I'm thinking about an attack that would send multiple requests and would abort them at this precise time which cause for skipping the rate limiter and ignore its purpose.